doc/sphinx/proof-engine/tactics.rst

```diff
@@ -734,6 +734,11 @@ Applying theorems
       are referred by names and non-dependent products by increasing numbers (see
       :ref:`bindings`).
 
+      .. flag:: Apply With Renaming
+
+         When on, this flag causes the names in the :n:`@term`'s type to be renamed for uniqueness.
+         By default no renaming is done.
```

Comment:
Maybe add some wording to say that this is a compatibility flag with the behavior pre-8.15. Otherwise, it's hard to understand why we would provide such an option.